packages/netlify-cms-backend-github/src/queries.js

```diff
@@ -0,0 +1,241 @@
+import gql from 'graphql-tag';
+import * as fragments from './fragments';
+
+export const repoPermission = gql`
+  query repoPermission($owner: String!, $name: String!) {
+    repository(owner: $owner, name: $name) {
+      id
+      viewerPermission
+    }
+  }
+`;
+
+export const user = gql`
+  query {
+    viewer {
+      id
+      avatar_url: avatarUrl
+      name
+    }
+  }
+`;
+
+export const blob = gql`
+  query blob($owner: String!, $name: String!, $expression: String!) {
+    repository(owner: $owner, name: $name) {
+      id
+      object(expression: $expression) {
+        ... on Blob {
+          ...BlobWithTextParts
+        }
+      }
+    }
+  }
+  ${fragments.blobWithText}
+`;
+
+export const unpublishedBranchFile = gql`
```

Comment:
I find this very cool. In one query you can get the file from the `head` branch and check if the file exists on the `base` branch

Reply:
Wow that's crazy